go/types.go

```diff
@@ -342,3 +342,24 @@ type Precertificate struct {
 func (m *MerkleTreeLeaf) X509Certificate() (*x509.Certificate, error) {
 	return x509.ParseCertificate(m.TimestampedEntry.X509Entry)
 }
+
+type scterror int
+
+//Preallocate errors for performance
+var (
+	INVALID_VERSION   error = scterror(1)
+	SCT_TOO_LARGE     error = scterror(2)
+	NOT_ENOUGH_BUFFER error = scterror(3)
+)
+
+func (e scterror) Error() string {
+	switch e {
+	case INVALID_VERSION:
+		return "Unable to serialize SignedCertificateTimestamp: Invalid version"
+	case SCT_TOO_LARGE:
+		return "Unable to serialize SignedCertificateTimestamp: SCT too large"
+	case NOT_ENOUGH_BUFFER:
+		return "Unable to serialize SignedCertificateTimestamp: Not enough buffer"
+	}
+	return "Unknown error occured in sct"
```

Comment:
/sct/SCT/ for consistency, also maybe add "serialisation" to the end, otherwise it sounds like the SCT is broken somehow.